index.js

```diff
@@ -42,17 +41,34 @@ const register = function(server, options) {
 
   server.decorate('request', 'timingEnd', function(key) {
     const timing = this.plugins['hapi-timing'][key];
-    timing.end = new Date();
-    timing.elapsed = timing.end - timing.start;
+    this.plugins['hapi-timing'][key].end = new Date();
+    this.plugins['hapi-timing'][key].elapsed = timing.end - timing.start;
   });
 
+  if (options.requestLifecycle) {
+    const eventNames = ['onRequest', 'onPreAuth', 'onPostAuth', 'onPreHandler', 'onPostHandler'];
+    eventNames.forEach((eventName, i) => {
+      server.ext(eventName, (request, h) => {
+        if (i !== 0) {
+          // call timingEnd on previous event:
+          request.timingEnd(eventNames[i - 1]);
+        }
+        request.timingStart(eventName);
+        return h.continue;
+      });
+    });
+  }
   server.events.on('response', (request) => {
+    // turn off last event handler if we were monitoring each lifecycle segment:
+    if (options.requestLifecycle) {
+      request.timingEnd('onPostHandler');
+    }
     // check the tail response times and notify if needed:
     const responseTime = request.info.responded - request.info.received;
     const plugin = request.route.settings.plugins['hapi-timing'];
     const threshold = (plugin && plugin.threshold) ? plugin.threshold : options.threshold;
-    if (responseTime > threshold) {
-      requestTimeoutExpired(responseTime, threshold, request);
+    if (options.requestLifecycle || responseTime > threshold) {
```

Comment:
requestLifecycle shouldn't trigger logs all the time, verbose mode should - speaking of that, where is that code in here?